app/src/main/java/com/codefororlando/petadoption/view/LocationDialogFragment.java

```diff
@@ -30,6 +32,8 @@
 
     private static final int LOCATION_PERMISSION_REQUEST_CODE = 1;
 
+    private PublishSubject dismissalSubject = PublishSubject.create();
```

Comment:
The local type should just be `Subject`. Also, you are missing the generic which appears to be a Boolean however it doesn't appear that anything other than `true` is passed meaning you are using the subject as a signal bus. In RXv1 you could use `Void` as the type, however, RXv2 does not permit that in Java, however, Kotlin can use `Unit`.

Considering the above I would suggest either converting this class to kotlin so that `Unit` can be used or creating an enum signal if you want to leave this class as Java.

Java approach
```java
public enum Dismissal { Requested; }

private PublishSubject<Dismissal> dismissalSubject = PublishSubject.create();
```